### DISCUSSION TITLE
Checking for Convergence / Handling Divergences

### DISCUSSION THREAD

**Discussion by PK1706** [pymc-labs/pymc-marketing#781] [Q&A] [answered]
Hello,
since PyMC-Marketing is my first step into bayesian statistics, deciding whether my MMM is trustworthy / converged is quite hard to do. 
While the model results look quite promising and not to unrealistic I almost always get around 2 to 20 divergences (target accept already being set to 0.99). In what way should I respond to this? Use the model regardless or try to improve the model?
I have read Bayesian Analysis with Python by Osvaldo Martin as well as countless other articles, but am still a bit lost.
What is often mentioned is that a reparametrization of the model can help with convergence, this is however 1) not trivial, and 2) not really possible when using PyMC-Marketing, where the model itself is wrapped and not accessible by the user. 

I have tried changing priors (although I have read this isn't really best practice) to combat this, but didn't get satisfactory results.
The only thing that did reduce and in some runs eliminate divergences was using a student-t likelihood. I am however unsure about how to choose the right prior for the normality parameter.

Are there any solutions to reducing divergences that I am missing or are divergences not as big of a deal as some make it seem?
I would highly appreciate some advice on this and possibly a rule of thumb step-by-step approach.
Thanks a lot!

**Comment by AlfredoJF** [comment, marked as answer]
Hi @PK1706 

Have a look at these [Prior Choice Recommendations](https://github.com/stan-dev/stan/wiki/Prior-Choice-Recommendations#default-prior-for-treatment-effects-scaled-based-on-the-standard-error-of-the-estimate) and see if you can decrease the number of divergences by selecting better priors for your use case.

Also, try to increase the default params `chains`, `draws` and `tune` in the `fit` method.

```python
mmm.fit(X=X_train,
        y=y_train,
        target_accept=0.95,
        chains=6,
        random_seed=123,
        draws=3_000, 
        tune=2_000, 
        )
```

**Reply by PK1706** [reply]
Thanks, tuning prior distributions a bit already reduced divergences in most cases!
